benches/parallel.rs

```diff
@@ -0,0 +1,341 @@
+#![feature(test)]
+
+extern crate cgmath;
+extern crate rand;
+extern crate specs;
+
+extern crate test;
+
+use cgmath::Vector2;
+use rand::thread_rng;
+use specs::prelude::*;
+use specs::storages::NullStorage;
+
+use test::Bencher;
+
+type Vec2 = Vector2<f32>;
+
+// -- Components --
+
+#[derive(Clone, Copy, Debug)]
+struct Pos(Vec2);
+
+impl Component for Pos {
+    type Storage = VecStorage<Pos>;
+}
+
+#[derive(Clone, Copy, Debug)]
+struct Vel(Vec2);
+
+impl Component for Vel {
+    type Storage = VecStorage<Vel>;
+}
+
+#[derive(Clone, Copy, Debug)]
+struct Force(Vec2);
+
+impl Component for Force {
+    type Storage = VecStorage<Force>;
+}
+
+#[derive(Clone, Copy, Debug)]
+struct InvMass(f32);
+
+impl Component for InvMass {
+    type Storage = VecStorage<InvMass>;
+}
+
+#[derive(Clone, Copy, Debug)]
+struct Lifetime(f32);
+
+impl Component for Lifetime {
+    type Storage = VecStorage<Lifetime>;
+}
+
+#[derive(Clone, Copy, Debug)]
+struct Ball {
+    radius: f32,
+}
+
+impl Component for Ball {
+    type Storage = VecStorage<Ball>;
+}
+
+#[derive(Clone, Copy, Debug)]
+struct Rect {
+    a: f32,
+    b: f32,
+}
+
+impl Component for Rect {
+    type Storage = VecStorage<Rect>;
+}
+
+#[derive(Clone, Copy, Debug)]
+enum Spawner {
+    Ball { radius: f32, inv_mass: f32 },
+    Rect { a: f32, b: f32, inv_mass: f32 },
+}
+
+impl Component for Spawner {
+    type Storage = HashMapStorage<Spawner>;
+}
+
+#[derive(Clone, Copy, Debug)]
+struct SpawnRequests(usize);
+
+impl Component for SpawnRequests {
+    type Storage = HashMapStorage<SpawnRequests>;
+}
+
+#[derive(Clone, Copy, Debug)]
+struct Collision {
+    a: Entity,
+    b: Entity,
+    contact: Vec2,
+}
+
+impl Component for Collision {
+    type Storage = DenseVecStorage<Collision>;
+}
+
+#[derive(Clone, Copy, Debug)]
+struct Room {
+    inner_width: f32,
+    inner_height: f32,
+}
+
+impl Component for Room {
+    type Storage = HashMapStorage<Room>;
+}
+
+#[derive(Clone, Copy, Debug)]
+enum Color {
+    Green,
+    Red,
+}
+
+impl Component for Color {
+    type Storage = VecStorage<Color>;
+}
+
+#[derive(Clone, Copy, Debug, Default)]
+struct KillsEnemy;
+
+impl Component for KillsEnemy {
+    type Storage = NullStorage<KillsEnemy>;
+}
+
+// -- Resources --
+
+#[derive(Clone, Copy, Debug)]
+struct DeltaTime(f32);
+
+impl Resource for DeltaTime {}
```

Comment:
Imo the compiler output looks quite reasonable (https://is.gd/4wDZDx):
```Rust
error[E0277]: the trait bound `Foo: std::fmt::Debug` is not satisfied
 --> <anon>:4:5
  |
4 |     add(Foo);
  |     ^^^ the trait `std::fmt::Debug` is not implemented for `Foo`
  |
  = note: `Foo` cannot be formatted using `:?`; if it is defined in your crate, add `#[derive(Debug)]` or manually implement it
  = note: required because of the requirements on the impl of `Resource` for `Foo`
  = note: required by `add`
```
I'm not sure what the others think about it but I would prefer a blanket implementation.